lib/parse.js

```diff
@@ -1278,6 +1278,274 @@ function parse($TEXT, options) {
+                    token_error(duplicate, "SyntaxError: Parameter " + duplicate.value + " was used already");
+                }
+            }
+        };
+
+        return tracker;
+    }
+
+    function parameters() {
+        var start = S.token;
+        var first = true;
+        var params = [];
+        var used_parameters = track_used_binding_identifiers(true, S.input.has_directive("use strict"));
+
+        expect("(");
+
+        while (!is("punc", ")")) {
+            if (first) {
+                first = false;
+            } else {
+                expect(",");
+            }
+
+            var param = parameter(used_parameters);
+            params.push(param);
+
+            if (param instanceof AST_Expansion) {
+                break;
+            }
+        }
+
+        next();
+        return params;
+    }
+
+    function parameter(used_parameters, symbol_type) {
+        var param;
+        var expand = false;
+        used_parameters = used_parameters === undefined ?
+                          track_used_binding_identifiers(true, S.input.has_directive("use strict")) : used_parameters;
+        if (is("expand", "...")) {
+            expand = S.token;
+            used_parameters.mark_spread(S.token);
+            next();
+        }
+        param = binding_element(used_parameters, symbol_type);
+
+        if (is("operator", "=") && expand === false) {
+            used_parameters.mark_default_assignment(S.token);
+            next();
+            param.default = expression(false);
+        }
+
+        if (expand !== false) {
+            if (!is("punc", ")")) {
+                unexpected();
+            }
+            param = new AST_Expansion({
+                start: expand,
+                expression: param,
+                end: expand
+            });
+        }
+        used_parameters.check_strict();
+
+        return param;
+    }
+
+    function binding_element(used_parameters, symbol_type) {
+        var elements = [];
+        var first = true;
+        var is_expand = false;
+        var expand_token;
+        var first_token = S.token;
+        used_parameters = used_parameters === undefined ?
+                          track_used_binding_identifiers(false, S.input.has_directive("use strict")) : used_parameters;
+        symbol_type = symbol_type === undefined ? AST_SymbolFunarg : symbol_type;
+        if (is("punc", "[")) {
+            next();
+            while (!is("punc", "]")) {
+                if (first) {
+                    first = false;
+                } else {
+                    expect(",");
+                }
+
+                if (is("expand", "...")) {
+                    is_expand = true;
+                    expand_token = S.token;
+                    used_parameters.mark_spread(S.token);
+                    next();
+                }
+                if (is("punc")) {
+                    switch (S.token.value) {
+                      case ",":
```

Comment:
Actually it should be fine (see elision at the start of the loop). But there were no tests for this.